src/plugins/intel_cpu/src/nodes/fullyconnected.cpp

```diff
@@ -265,6 +265,13 @@ void FullyConnected::getSupportedDescriptors() {
     }
 }
 
+void FullyConnected::createPrimitive() {
+    setPostOps(attr, outDims);
```

Comment:
`outDims` may be a dummy shape in dynamic model, is that a problem when a binary post Ops is being fused into FullyConnected ?

maybe it's not a problem for FullyConnected,  since OC is always static and so far we only use `outDims` in setPostOps when fusing per-OC post-Ops.

If so, why not take one more step, when the weights for convolution is also constant and the OC is static, we can also do this optimization which benefits more, what do you think?

Reply:
On the high level, yes, using `outDims` being a "dummy" shape is rather dangerous, since the `DnnlPostOpsComposer` requires the output shape to be equal to the output shape. But, on the FC operation level we know that the post operations may only depend on the channel size. So yes, this is a coupling, but staying on such a low level we can sacrifice the coupling for better performance. At least such coupling is bounded only on the node implementation level and does not impact anything outside it.
From the implementation behavior, following https://github.com/openvinotoolkit/openvino/blob/0c9abf43a91fbe2dc95974b6cb2651bd883d1b12/src/plugins/intel_cpu/src/dnnl_postops_composer.cpp#L28-L29 we see that only `OC` dimension is used for the binary post ops, all others are always 1. So that should work. And in opposite, if it didn't work, that would mean that something was wrong with the `DnnlPostOpsComposer`, since it requires dependency on the other dims when it is meaningless.
As for `Convolution`. We already have the very same optimization, but implemented in slightly different fashion, please take a look at https://github.com/openvinotoolkit/openvino/blob/0c9abf43a91fbe2dc95974b6cb2651bd883d1b12/src/plugins/intel_cpu/src/nodes/conv.cpp#L1309-L1318 the only difference is that, due to some comprehensive fusing handling, we still have to call the attributes initialization under some conditions.

Reply:
Yes, indeed, from oneDNN API binary postOps can support any shape with broadcast semantic that may depending on exact dimension value other than OC, but we didn't observe such use case for now, so in theory we can re-build attr only when OC is changed, this may also apply to matmul.  for example, in matmul, the batch dimension may be dynamic but the channel dimension is usually static. this method can be generalized, maybe next time.

Reply:
I'm just wondering, maybe it does makes sense to reflect such a property in the `DnnlPostOpsComposer` constructor? I mean add a constructor which takes OC dimension only, so that to ensure that all other dims are not used?